### ISSUE TITLE
Error with 2D array input

### ISSUE THREAD

**Issue by ashedw88** [deeplearning4j/deeplearning4j#739]
I receive an exception when I attempt to input a 2D array created in the following way:

INDArray input = Nd4j.create(x); where x is a 2D array: double[100][100]
network.output(input);

// Exception
java.lang.IllegalArgumentException: Invalid input: expect output columns must be equal to rows 100 x columns 100 but was instead [100, 100]
	at org.deeplearning4j.nn.conf.preprocessor.FeedForwardToCnnPreProcessor.preProcess(FeedForwardToCnnPreProcessor.java:84)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.activationFromPrevLayer(MultiLayerNetwork.java:483)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForward(MultiLayerNetwork.java:546)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForward(MultiLayerNetwork.java:532)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:1424)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:1405)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.output(MultiLayerNetwork.java:1440)

**Comment by agibsonccc** [comment]
Please include the network config

**Comment by ashedw88** [comment]
MultiLayerConfiguration.Builder builder = new NeuralNetConfiguration.Builder()
		                 .seed(seed)
		                 .iterations(iterations)
		                 .batchSize(batchSize)
		                 .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
		                 .constrainGradientToUnitNorm(true)
		                 .l2(2e-4)
		                 .regularization(true)
		                 .useDropConnect(true)
			                .list(4)
			                .layer(0, new ConvolutionLayer.Builder(8, 8) //16 filters kernel size 8 stride 4
			                        .stride(4, 4)
			                        .nOut(16)
			                        .dropOut(0.5)
			                        .activation("relu")
			                        .weightInit(WeightInit.XAVIER)
			                        .build())
			                .layer(1, new ConvolutionLayer.Builder(4, 4) //32 filters kernel size 4 stride 2
			                		.stride(2, 2)
			                        .nOut(32)
			                        .dropOut(0.5)
			                        .activation("relu")
			                        .weightInit(WeightInit.XAVIER)
			                        .build())
			                .layer(2, new DenseLayer.Builder() //fully connected with 256 rectified units
			                        .nOut(256)
			                        .activation("relu")
			                        .weightInit(WeightInit.XAVIER)
			                        .dropOut(0.5)
			                        .build())
			                .layer(3, new OutputLayer.Builder(LossFunction.SQUARED_LOSS) //output layer
			                        .nOut(numActions)
			                        .weightInit(WeightInit.XAVIER)
			                        .activation("softmax")
			                        .build())

		                 .backprop(true).pretrain(false);

		        new ConvolutionLayerSetup(builder,numRows,numColumns,1);

**Comment by agibsonccc** [comment]
Thanks!

**Comment by agibsonccc** [comment]
https://github.com/deeplearning4j/deeplearning4j/pull/744
